components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1113,10 +1115,13 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 			cancelReq()
 		}()
 
-		err = m.manager.markWorkspace(ctx, workspaceID, addMark(startedDisposalAnnotation, util.BooleanTrueString))
-		if err != nil {
-			tracing.LogError(span, err)
-			log.WithError(err).Error("was unable to update pod's start disposal state - this might cause an incorrect disposal state")
+		if disposalStatus.Status != "" {
+			disposalStatus.Status = DisposalStarted
+			err = m.manager.markDisposalStatus(ctx, workspaceID, disposalStatus)
+			if err != nil {
+				tracing.LogError(span, err)
+				log.WithError(err).Error("was unable to update pod's start disposal status - this might cause an incorrect disposal status")
+			}
```

Comment:
I think this part does not need, correct? Because we have this already https://github.com/gitpod-io/gitpod/blob/795b44e1d7e463394b080d5ee299a0fcae5a508f/components/ws-manager/pkg/manager/monitor.go#L1292-L1299